eos-extra-target-generator

```diff
@@ -0,0 +1,14 @@
+#!/bin/sh
+
+normal_dir="$1"
+target_unit="${1}/eos-extra-settled.target"
+
+cat >"${target_unit}" <<EOF
```

Comment:
Probably overly pedantic, but I'm a little worried about writing this file directly and then editing it below. Both because we might error before it completes and that systemd might act on the unit before it's complete. I think it would be better to use a temp file and then copy/move into place at the end.

Reply:
I don't think systemd acts on the unit before the generator exits. From the manpage:
`If you are careful, you can implement generators in shell scripts. We do recommend C code however, since generators are executed synchronously and hence delay the entire boot if they are slow.
`

I'm not sure what you mean by "we might error before it completes", but it that is a real problem we may need to check whether https://github.com/endlessm/eos-boot-helper/blob/master/dracut/image-boot/eos-image-boot-generator#L79-L98 is also affected.

If we use a temp file which disappears before the generator exits I don't know if systemd would complain when trying to remove it on deamon-reload [1], or if it only keeps track of the files left by the generator after it terminates.

[1] `Units written by generators are removed when the configuration is reloaded. That means the lifetime of the generated units is closely bound to the reload cycles of systemd itself.`

Reply:
@jprvita For the "completes before error case", I mean if the script exited because of an error before it could write the `RequiresMountsFor` line. But since this script isn't running with `-e`, I guess that's really unlikely.

Thanks for checking the blocking behavior of the generator. In that case I guess it's fine to edit in place.

If you did want to use a temp file, there are completely standard ways to do it in shell to ensure it gets cleaned up. The temp file would only be used while editing and would have no connection to the actual file copied into /etc. But given how generators work, it's probably fine as is.